inc/class-core-sitemaps-taxonomies.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * Taxonomies sitemap.
+ *
+ * @package Core_Sitemaps
+ */
+
+/**
+ * Class Core_Sitemaps_Taxonomies.
+ * Builds the sitemap pages for Taxonomies.
+ */
+class Core_Sitemaps_Taxonomies extends Core_Sitemaps_Provider {
+	/**
+	 * Core_Sitemaps_Taxonomies constructor.
+	 */
+	public function __construct() {
+		$this->object_type = 'taxonomy';
+		$this->route       = '^sitemap-taxonomies-([A-z]+)-?([0-9]+)?\.xml$';
+		$this->slug        = 'taxonomies';
+	}
+
+	/**
+	 * Produce XML to output.
+	 */
+	public function render_sitemap() {
+		global $wp_query;
+
+		$sitemap  = get_query_var( 'sitemap' );
+		$sub_type = get_query_var( 'sub_type' );
+		$paged    = get_query_var( 'paged' );
+
+		$sub_types = $this->get_object_sub_types();
+
+		if ( ! isset( $sub_types[ $sub_type ] ) ) {
+			// Invalid sub type.
+			$wp_query->set_404();
+			status_header( 404 );
+
+			return;
+		}
+
+		$this->sub_type = $sub_types[ $sub_type ]->name;
+		if ( empty( $paged ) ) {
+			$paged = 1;
+		}
+
+		if ( $this->slug === $sitemap ) {
+			$url_list = $this->get_url_list( $paged );
+			$renderer = new Core_Sitemaps_Renderer();
+			$renderer->render_sitemap( $url_list );
+
+			exit;
+		}
+	}
+
+	/**
+	 * Get a URL list for a taxonomy sitemap.
+	 *
+	 * @param int $page_num Page of results.
+	 *
+	 * @return array $url_list List of URLs for a sitemap.
+	 */
+	public function get_url_list( $page_num = 1 ) {
+
+		$type = $this->sub_type; // Find the query_var for sub_type.
+		if ( empty( $type ) ) {
+			$type = $this->object_type; // If empty set to object_type instead.
+		}
+
+		// Get all of the taxonomies that are registered.
+		$taxonomies = $this->get_object_sub_types();
+
+		$url_list = array();
+
+		foreach ( $taxonomies as $taxonomy ) {
+			// if the query_var matches a taxonomy name, get the terms for that tax.
+			if ( $type === $taxonomy->name ) {
+
+				$taxonomy_terms = get_terms(
+					array(
+						'fields'     => 'ids',
+						'taxonomy'   => $taxonomy->name,
+						'orderby'    => 'term_order',
+						'hide_empty' => true,
+					)
+				);
+
+				// Loop through the terms and get the latest post stored in each.
+				foreach ( $taxonomy_terms as $term ) {
+
+					$last_modified = get_posts(
+						array(
+							'tax_query'      => array(
```

Comment:
This is causing `WARNING | Detected usage of tax_query, possible slow query.` in phpcs so may need changing to something more performant?

Reply:
Probably need to improve general performance here as well based on: https://github.com/GoogleChromeLabs/wp-sitemaps/pull/55